plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTestHelper.java

```diff
@@ -54,6 +57,22 @@ public static Dictionary createAssetPackLocationsDictionary() {
     return testDict;
   }
 
+  public static void assertMockAssetPackExceptionDictionaryIsExpected(
+      Dictionary mockExceptionDictionary, String expectedMessage, int expectedErrorCode) {
+    // when calling get class methods on Mocked objects, the returned named will contain Mockito
+    // specific suffix such as
+    // com.google.android.play.core.assetpacks.AssetPackException$$EnhancerByMockitoWithCGLIB$$c5.
+    // Hence we need to apply string operations to only take the prefix.
+    String testExceptionType = (String) mockExceptionDictionary.get("type");
+    testExceptionType =
+        testExceptionType.substring(0, AssetPackException.class.getCanonicalName().length());
```

Comment:
If the testExceptionType string is shorter than AssetPackException.class.getCanonicalName() you would get an IndexOutOfBoundsException here. Rather than getting a substring of testExceptionType, could you split it by `$$` characters and take the first element the returned array?

Reply:
Used split by ```$$``` approach